### ISSUE TITLE
Finalize Async progress

### ISSUE THREAD

**Issue by zbeekman** [sourceryinstitute/OpenCoarrays#117]
For 2.x release

Due diligence to attempt before 2.x release of async progress features
- [ ] Verify that all the async progress logic is inside `devel` branch (Tasked to @afanfa)
- [ ] Add advanced option to enable from CMake.
- ~~Test across at least 2 different nodes on other platforms than Cray and Mellanox, the simpler/dumber the better.... Try openlab.umiacs.umd (@fanfa send @zbeekman a good test case, please)... get test case from Alessandro~~

**Comment by rouson** [comment]
@afanfa, please update us on whether is ready for release.

**Comment by zbeekman** [comment]
I'm going to attempt to test using cluster here at UMD, per @afanfa's instructions

**Comment by zbeekman** [comment]
Also, waiting on @afanfa's input on the [progress for events branch](https://github.com/sourceryinstitute/opencoarrays/tree/progress-for-events) as to whether or not that needs to be merged into master or devel.

**Comment by zbeekman** [comment]
@afanfa said (via email):

> Hi,
Please do not delete the events branch.
Feel free to delete the progress - for- events.
Thanks

**Comment by zbeekman** [comment]
The machine that I was planning on testing this on is no longer available to me. Should we close this issue and just assume that this works? Or does anyone know a good system to test on?

**Comment by afanfa** [comment]
If I recall correctly, we decided to include this feature in the next release as experimental (disabled by default). This feature is really useful to understand whether the application can get some big advantage from computation/communication overlapping or not, but it is not really the best option for production codes (uses non-standard ways to guarantee MPI progress).

**Comment by zbeekman** [comment]
OK, so long as you can confirm that all the logic is in the devel branch, then I'll leave this open and add an (advanced) CMake option to enable this feature on devel.

**Comment by zbeekman** [comment]
Closing due to abandonment of devel... Should we consider porting this feature to master, @afanfa and @rouson or should we abandon this experimental feature as well?
